contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -0,0 +1,147 @@
+use axelar_wasm_std::msg_id::HexTxHashAndEventIndex;
+use axelar_wasm_std::nonempty;
+use cosmwasm_std::{
+    Addr, Api, Event, HexBinary, QuerierWrapper, Response, Storage, Uint256, WasmMsg,
+};
+use error_stack::{report, Result, ResultExt};
+use router_api::client::Router;
+use router_api::{Address, ChainName, CrossChainId, Message};
+use sha3::{Digest, Keccak256};
+
+use crate::contract::Error;
+use crate::events::AxelarnetGatewayEvent;
+use crate::executable::AxelarExecutableClient;
+use crate::state::{self};
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn call_contract(
+    store: &mut dyn Storage,
+    block_height: u64,
+    router: &Router,
+    chain_name: ChainName,
+    sender: Addr,
+    destination_chain: ChainName,
+    destination_address: Address,
+    payload: HexBinary,
+) -> Result<Response, Error> {
+    let counter = state::increment_msg_counter(store).change_context(Error::InvalidStoreAccess)?;
+
+    // TODO: Retrieve the actual tx hash from core, since cosmwasm doesn't provide it. Use the block height as the placeholder in the meantime.
+    let message_id = HexTxHashAndEventIndex {
+        tx_hash: Uint256::from(block_height).to_be_bytes(),
```

Comment:
oh well this is not as clean as using the blockhash, but oh well

Reply:
Yeah, weird that they haven't exposed blockhash / txhash in cosmwasm yet. Regardless, this will be changed soon